metashare/api/models.py

```diff
@@ -189,35 +200,119 @@ class Project(mixins.HashIdMixin, mixins.TimestampsMixin, SlugMixin, models.Mode
+        # TODO: revisit this?
+        return True
+
     def save(self, *args, **kwargs):
-        if not self.branch_name:
-            self.branch_name = slugify(self.name)
         super().save(*args, **kwargs)
 
+        if self.tracker.has_changed("branch_name"):
+            self.notify_has_branch_name()
+
+    def notify_has_branch_name(self):
+        from .serializers import ProjectSerializer
+
+        payload = ProjectSerializer(self).data
+        message = {"type": "PROJECT_UPDATE", "payload": payload}
+        async_to_sync(push.push_message_about_instance)(self, message)
+
     class Meta:
         ordering = ("-created_at", "name")
-        unique_together = (("name", "repository"), ("branch_name", "repository"))
+        unique_together = (("name", "repository"),)
 
 
 class TaskSlug(AbstractSlug):
     parent = models.ForeignKey("Task", on_delete=models.PROTECT, related_name="slugs")
 
 
 class Task(mixins.HashIdMixin, mixins.TimestampsMixin, SlugMixin, models.Model):
+    tracker = FieldTracker(fields=("branch_name",))
+
     name = models.CharField(max_length=50)
     project = models.ForeignKey(Project, on_delete=models.PROTECT, related_name="tasks")
     description = MarkdownField(blank=True, property_suffix="_markdown")
     assignee = models.ForeignKey(
         User, on_delete=models.SET_NULL, null=True, related_name="assigned_tasks"
     )
+    branch_name = models.SlugField(max_length=50, null=True)
 
     slug_class = TaskSlug
 
+    def __str__(self):
+        return self.name
+
+    def subscribable_by(self, user):  # pragma: nocover
+        # TODO: revisit this?
+        return True
+
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+
+        if self.tracker.has_changed("branch_name"):
+            self.notify_has_branch_name()
+
+    def notify_has_branch_name(self):
+        from .serializers import TaskSerializer
+
+        payload = TaskSerializer(self).data
+        message = {"type": "TASK_UPDATE", "payload": payload}
+        async_to_sync(push.push_message_about_instance)(self, message)
+
     class Meta:
         ordering = ("-created_at", "name")
         unique_together = (("name", "project"),)
 
 
+SCRATCH_ORG_TYPES = Choices("Dev", "QA")
+
+
+class ScratchOrg(mixins.HashIdMixin, mixins.TimestampsMixin, models.Model):
+    tracker = FieldTracker(fields=("url",))
+
+    task = models.ForeignKey(Task, on_delete=models.PROTECT)
+    org_type = StringField(choices=SCRATCH_ORG_TYPES)
+    owner = models.ForeignKey(User, on_delete=models.PROTECT)
+    last_modified_at = models.DateTimeField(null=True)
+    expires_at = models.DateTimeField(null=True)
+    latest_commit = StringField(blank=True)
+    latest_commit_url = models.URLField(blank=True)
+    url = models.URLField(null=True)
+    has_changes = models.BooleanField(default=False)
+
+    def subscribable_by(self, user):  # pragma: nocover
+        # TODO: revisit this?
+        return True
```

Comment:
Probably okay for now? Any user subscribes to this if they load the page while the org is still being created.